controllers/city.controller.go

```diff
@@ -9,25 +9,26 @@ import (
 	"project_weather/repositories"
 )
 
-type CityDB interface {
-	FindCityByID(id string) (*model.City, error)
-	RegisterCity(city *model.City) error
-	UpdateCity(city *model.City) error
-	DeleteCityByID(id string) error
-	GetAllCity() ([]model.City, error)
+type CityDBController interface {
+	GetRoutes() []Route
+	GetCityById(ctx *fiber.Ctx) error
+	DeleteCity(ctx *fiber.Ctx) error
+	RegisterCity(ctx *fiber.Ctx) error
+	UpdateCity(ctx *fiber.Ctx) error
+	GetAllCities(ctx *fiber.Ctx) error
 }
 
-type CityController struct {
-	DB repositories.CityRepo
+type cityController struct {
```

Comment:
Nice! Making this private (by decapitalizing it in golang) and exporting the interface encapsulates the class in the package